pkg/mqtt5/client.go

```diff
@@ -9,43 +9,31 @@ import (
 	"crypto/x509"
 	"errors"
 	"fmt"
-
 	"net"
 	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/eclipse/paho.golang/paho"
 
 	"github.com/IOTechSystems/go-mod-edge-utils/pkg/bootstrap/interfaces"
+	"github.com/IOTechSystems/go-mod-edge-utils/pkg/bootstrap/secret"
 	"github.com/IOTechSystems/go-mod-edge-utils/pkg/log"
 	"github.com/IOTechSystems/go-mod-edge-utils/pkg/mqtt5/config"
-	"github.com/eclipse/paho.golang/paho"
 )
 
 const (
-	AuthModeNone             = "none"
-	AuthModeUsernamePassword = "usernamepassword"
-	AuthModeCert             = "clientcert"
-	AuthModeCA               = "cacert"
-
-	SecretUsernameKey = "username"
-	SecretPasswordKey = "password"
-	SecretClientKey   = "clientkey"
-	SecretClientCert  = AuthModeCert
-	SecretCACert      = AuthModeCA
+	DefaultDialTimeOut = 30
 )
 
 type Mqtt5Client struct {
 	configuration config.Mqtt5Config
-	authData      SecretData
+	authData      secret.SecretData
 	mqtt5Client   *paho.Client
 	connect       *paho.Connect
 	isConnected   bool
-}
-
-type SecretData struct {
-	Username     string
-	Password     string
-	KeyPemBlock  string
-	CertPemBlock string
-	CaPemBlock   string
+	mutex         *sync.RWMutex
```

Comment:
use Mutex instead of RWMutex, because it is not for read/write

Reply:
updated